aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -2815,6 +2815,61 @@ TORCH_IMPL_FUNC(linalg_vector_norm_out)(const Tensor& self, const Scalar& scalar
   // values larger than 10^53 (same for negative numbers), so that's fine.
   auto ord = scalar_ord.toDouble();
   auto dim = opt_dim.value_or(IntArrayRef{});
+  auto size = self.sizes();
+  auto ndim = self.dim();
+
+  bool all_reduction_dims_are_one_dimensional = true;
+  if (result.sizes().empty()){
+    // reduce over all dims and all dims are 1-dimensional
+    // if ndim == 0, self is scalar
+    for (const auto dim_ : c10::irange(ndim)) {
+      if (size[dim_] != 1){
+        all_reduction_dims_are_one_dimensional = false;
+        break;
+      }
+    }
+  } else {
+    // all reduction dims are 1-dimensional
+    if (opt_dim.has_value() && !opt_dim->empty()) {
+      DimVector dims_ = at::native::make_dim_vector(opt_dim, self.dim());
+      maybe_wrap_dims(dims_, self.dim());
+      for (const auto i : c10::irange(dims_.size())) {
+        size_t dim_ = dims_[i];
+        if (size[dim_] != 1){
+          all_reduction_dims_are_one_dimensional = false;
+          break;
+        }
+      }
+    } else {
```

Comment:
Pretty sure this is dead code, as if the result is not 0-dim, it must have `dim` specified.